### DISCUSSION TITLE
Unit test projects doesnt recognize project referenced with PrivateAssets="all"

### DISCUSSION THREAD

**Discussion by jeminpro** [teroneko/Teronis.DotNet#15] [Q&A] [answered]
The NuGet package "Teronis.MSBuild.Packaging.ProjectBuildInPackage" works well on adding the project references for dotnet pack, but the main issue is the unit test project which is referencing the project (i.e. the project which I am trying to package) doest recognize it has reference project when adding the attribute PrivateAssets="all". I am getting errors 

"Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0234	The type or namespace name '' does not exist in the namespace '' (are you missing an assembly reference?)

Is there a way to fix this?

**Comment by teneko** [comment, marked as answer]
One idea is to create another csproj, it may be a copy of your original but without PrivateAssets="all", and this is the `.csproj` you would reference in your test project.

**Comment by jonatanzazula** [comment]
Are there any other possibilities?

**Reply by teneko** [reply]
Yes there are other possibilities but they are too theoretical and strange that they are not usable. It has to be implemented on NuGet-side to support such scenarios. I am sorry. If you want some inspritation how strange a solution can get, then you can take a look at https://github.com/rabotcrypto/nethermind/tree/b8dbc067a5f0a1de2a5b3dc895495ab8c16185f8/rabot, but I advice strong against to use any pattern of it.
